### PR TITLE
use derived constants in kubernetes base

### PR DESCRIPTION
**Pull request by Tcharl**
contributes to #14235

---

Please make sure the below checklist is followed for Pull Requests.

- [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
- [x] Tests are added where necessary
- [x] [jhipster-online](https://github.com/jhipster/jhipster-online) was updated if necessary
- [x] Documentation is added/updated where necessary
- [x] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Comment by Tcharl** [comment]
I'm really not sure about this one, but hope the approach is right

**Comment by Tcharl** [comment]
I really suck at restructuring, and I don't really understand how to achieve that flattening of internal methods from another class. Any js blackbelt to help?

**Comment by mshima** [comment]
@Tcharl last commit you changed `end` from a function (1 task) to a getter (multiples tasks).
`_end()` should return an object.

**Comment by mshima** [comment]
I am thinking about adding snapshots test for deployment generators.
Current tests are not good enough.

**Comment by Tcharl** [comment]
Thank you vers much for your time @mshima , I'll one day become a js blackbelt because of your support! Fixing the issue now, what about the remaining change?

**Comment by Tcharl** [comment]
Triple checked: is it ok to go?

**Comment by Tcharl** [comment]
Hi everyone,

Back to that constants stuff ;-). Thank you very much one more time for the snapshot testing @mshima , it's  awesome for troubleshooting!

I think the new snapshots are only fixing existing bugs, can someone take a look please?

@deepu105 , I think that you're the one able to tell if the new version generated for the 'kubernetes helm: gateway, mysql, psql, mongodb, mariadb microservices' test is better than the old one.

**Comment by deepu105** [comment]
I'll try it out when i hv time

**Comment by mshima** [comment]
Glad it worked.
You forgot to update the snapshots.

**Comment by Tcharl** [comment]
Yes, it was totally wished: I wanted to wait for the review so that buddies will know what changed, then I'll update the snapshot if everything is validated

**Comment by Tcharl** [comment]
Much more confident with this round :-)

File: generators/azure-app-service/index.js [hunk 1]
```diff
@@ -484,16 +486,19 @@ which is free for the first 30 days`);
     return this._default();
   }
 
-  _loadPlatformConfig(config = _.defaults({}, this.jhipsterConfig, defaultConfig), dest = this) {
-    super.loadPlatformConfig(config, dest);
+  _computeDerivedConfig(config = _.defaults({}, this.jhipsterConfig, defaultConfig), dest = this) {
```
Comment (mshima) [inline generators/azure-app-service/index.js:489]: And move to after priorities, it's not a priority.
Or drop the method and put in the task.

File: generators/ci-cd/index.js [hunk 2]
```diff
@@ -194,9 +175,8 @@ module.exports = class extends BaseBlueprintGenerator {
     return this._configuring();
   }
 
-  _loadPlatformConfig(config = _.defaults({}, this.jhipsterConfig, defaultConfig), dest = this) {
-    super.loadPlatformConfig(config, dest);
-    dest.cicdIntegrationsSnyk = config.cicdIntegrations || [];
+  _loadCiCdConfig(config = _.defaults({}, this.jhipsterConfig, defaultConfig), dest = this) {
```
Comment (mshima) [inline generators/ci-cd/index.js:178]: Better move to after priorities to don't mix priority and api.

File: package-lock.json [hunk 3]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "generator-jhipster",
   "version": "7.1.0",
-  "lockfileVersion": 2,
+  "lockfileVersion": 1,
```
Comment (mshima) [inline package-lock.json:4]: Revert lock file

File: test/__snapshots__/docker-compose.spec.js.snap [hunk 4]
```diff
@@ -1088,6 +1088,11 @@ jhipster:
     authentication:
       jwt:
         base64-secret: SECRET--50
+management:
+  metrics:
+    export:
+      prometheus:
+        enabled: true
```
Comment (mshima) [inline test/__snapshots__/docker-compose.spec.js.snap:1095]: Looks good due to test description.

File: test/__snapshots__/docker-compose.spec.js.snap [hunk 5]
```diff
@@ -1641,27 +1659,6 @@ Launch all your infrastructure by running: \`docker-compose up -d\`.
       - SERVER_PORT=80
     ports:
       - \\"8080:8080\\"
-
-  jhipster-registry:
-    image: jhipster/jhipster-registry:v6.8.0
-    volumes:
-      - ./central-server-config:/central-config
```
Comment (mshima) [inline test/__snapshots__/docker-compose.spec.js.snap:1648]: With the test description, there is no way to know if it's correct.

File: generators/generator-defaults.js [hunk 6]
```diff
@@ -96,6 +96,12 @@ const translationDefaultConfig = {
   },
 };
 
+const kubernetesDefaultConfig = {
+  kubernetesServiceType: 'LoadBalancer',
+  monitoring: 'no',
+  istio: false,
+};
+
```
Comment (mshima) [inline generators/generator-defaults.js:104]: Thinking about modularization, I think we should move to the Kubernetes folder, like I did with init generator.
You can import here and reexport for a clean api, this is the new standard for esm/node.

**Review by mshima — COMMENTED** [review]
It looks good, just a few suggestions.

File: test/__snapshots__/docker-compose.spec.js.snap (hunk 5 above)
Comment (Tcharl) [reply test/__snapshots__/docker-compose.spec.js.snap:1648]: the pseudo and password are not filled compared to the snapshot that need eureka: looking at the 'configure' method of the docker compose generator, it's meaning the eureka is not selected.
Also, looking at the test, eureka is not mentionned in the yo-rc and configuration, and by default (according to the doc), serviceDiscovery is set to none

**Review by mshima — APPROVED** [review]
LGTM.
Merge at will.

**Comment by Tcharl** [comment]
Pending for too long: merging it, and pursuing the adventure after looking at the 'two phase generation' PR

**Pull request merged**